src/power_grid_model/utils.py

```diff
@@ -52,20 +59,29 @@ def get_dataset_scenario(dataset: BatchDataset, scenario: int) -> SingleDataset:
         The dataset for a specific scenario
     """
 
+    def _get_dense_scenario(arr: np.ndarray) -> np.ndarray:
+        return arr[scenario]
+
+    def _get_sparse_scenario(arr: np.ndarray, indptr: IndexPointer) -> np.ndarray:
+        return arr[indptr[scenario] : indptr[scenario + 1]]
+
     def _get_component_scenario(component_scenarios: BatchComponentData) -> np.ndarray:
-        # TODO(mgovers): update this with columnar scenario access
         if isinstance(component_scenarios, np.ndarray):
-            return component_scenarios[scenario]
+            return _get_dense_scenario(component_scenarios)
 
-        indptr = component_scenarios["indptr"]
-        data = component_scenarios["data"]
-        if isinstance(indptr, np.ndarray) and isinstance(data, np.ndarray):
-            return data[indptr[scenario] : indptr[scenario + 1]]
+        if "indptr" in component_scenarios:
+            indptr = component_scenarios["indptr"]
+            data = component_scenarios["data"]
+            if isinstance(indptr, np.ndarray):
```

Comment:
this stuff has happened soooooo often now, it's probably a good idea to create a utility function. this would make a lot of our code a lot simpler

```py
def _extract_indptr_data(data: SparseBatchData) -> tuple[np.ndarray, SingleArray | SingleColumnarData]:
    indptr = data["indptr"]
    sub_data = data["data"]

    if isinstance(indptr, np.ndarray):
        return indptr, data

    raise ValueError("some error message")
```

if you want to, you can even go as far as to make it a template thingy using `TypeVar` so that you can even deduce the return type of `data`, depending on whether it's a `SparseBatchData`, a `SparseBatchArray` or a `SparseBatchColumnarData`

Reply:
Should we do this in this PR? 
The final goal is to add more extraction functions where the checks happen. Then we can remove the big component_data_checks and all places where those checks are manually done. We had to keep them because of mypy. (@Jerry-Jinfeng-Guo ref to your other comment)

I have now added _extract_indptr_data, _extract_columnar_data, _extract_row_based_data as an example. They arent used anywhere as of yet.

Reply:
See https://github.com/PowerGridModel/power-grid-model/issues/749